portfolio/src/main/java/com/google/sps/servlets/SuggestServlet.java

```diff
@@ -35,29 +35,34 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    showResults(response, "");
+    showResults(response, "", "");
   }
   
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String userCategory = getParameter(request, "category", "");
+    String userSearch = getParameter(request, "search", "");
     if (userCategory.equals("")) {
-      String suggestion = getParameter(request, "suggestion-body", "");
-      String category = getParameter(request, "category-select", "");
-      long timestamp = System.currentTimeMillis();
-      Entity taskEntity = new Entity("Suggestion");
-      taskEntity.setProperty("suggest", suggestion);
-      taskEntity.setProperty("category", category);
-      taskEntity.setProperty("timestamp", timestamp);
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      datastore.put(taskEntity);
-      response.sendRedirect("/index.html");
+      if (userSearch.equals("")){  
+        String suggestion = getParameter(request, "suggestion-body", "");
+        String category = getParameter(request, "category-select", "");
+        long timestamp = System.currentTimeMillis();
+        Entity taskEntity = new Entity("Suggestion");
+        taskEntity.setProperty("suggest", suggestion);
+        taskEntity.setProperty("category", category);
+        taskEntity.setProperty("timestamp", timestamp);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(taskEntity);
+        response.sendRedirect("/index.html");
+      } else {
+        showResults(response, "", userSearch);
+      }
     } else {
-       showResults(response, userCategory);
+       showResults(response, userCategory, "");
     }
   }
 
-  public void showResults(HttpServletResponse response, String sugCategory) throws IOException {
+  public void showResults(HttpServletResponse response, String sugCategory, String sugSearch) throws IOException {
```

Comment:
Consider using Optional if they could be empty

Reply:
I made the parameters null but I'm not sure if that's what you meant. From what I've read, the Optional class in java still doesn't allow you to pass in empty parameters but I could be looking at the wrong thing.

Reply:
You can have the method signature be
```
public void showResults(HttpServletResponse response, Optional<String> sugCategory, Optional<String> sugSearch)
```

And if you'd like to pass in empty Strings, you can do:

```
showResults(response, Optional.empty(), Optional.empty())
```